src/Mailgun/Resource/Api/Domains/CredentialListResponse.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * Copyright (C) 2013-2016 Mailgun
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+namespace Mailgun\Resource\Api\Domains;
+
+use Mailgun\Resource\CreatableFromArray;
+
+/**
+ * @author Sean Johnson <sean@mailgun.com>
+ */
+class CredentialListResponse implements CreatableFromArray
+{
+    /**
+     * @var int
+     */
+    private $totalCount;
+
+    /**
+     * @var Credential[]
+     */
+    private $items;
+
+    /**
+     * @param array $data
+     *
+     * @return CredentialListResponse
+     */
+    public static function createFromArray(array $data)
+    {
+        $items = [];
+
+        foreach ($data['items'] as $item) {
+            $items[] = new Credential(
+                $item['size_bytes'],
+                new \DateTime($item['created_at']),
+                $item['mailbox'],
+                $item['login']
```

Comment:
Interesting question you bring upp here. Should we trust the API response or not?

Should we blindly assume the API does as the doc says or should we verify that at each request?